<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN">
<html>
<head>
<title>MOEDivCon Integration Help</title>
<meta http-equiv="content-type" content="text/html; charset=utf-8" />
<meta http-equiv="content-style-type" content="text/css" />
<style>
h1 {font: bold 2.0em Helvetica, sans-serif; color: #0099DD;
    text-align: center;}
h2 {font: bold 1.5em Helvetica, sans-serif; color: #44AAEE;
    background-color: #FFFFFF; counter-increment:h2n;counter-reset:h3n}
h2:before {content: counter(h2n, upper-roman) ".  "}
h3 {font: bold 1.25em Helvetica, sans-serif; color: #44AAEE;
    background-color: #FFFFFF;counter-increment:h3n;counter-reset:h4n}
h3:before {content: counter(h3n, upper-latin) ".  "}
h4 {font: bold 1.00em Helvetica, sans-serif; color: #0099DD;
    background-color: #FFFFFF;counter-increment:h4n;counter-reset:h5n}
h4:before {content: counter(h4n, decimal) ".  "}
h5 {font: bold 1.00em Helvetica, sans-serif; color: #0099DD;
    background-color: #FFFFFF;counter-increment:h5n;}
h5:before {content: counter(h5n, lower-latin) ".  "}
body {font: 12pt "Times New Roman", serif; margin: 3em;
    background-color: #F0F7FF;}
ol  {list-style-type: decimal;}
ol ol {list-style-type: lower-latin;}
ol#tocl {list-style-type: upper-roman;}
ol ol#tocl {list-style-type: upper-latin;}
ol ol ol#tocl {list-style-type: decimal}
th.white {font: bold 2.0em Helvetica, sans-serif; color: white}
th.green {font: bold 2.0em Helvetica, sans-serif; color: green}
th.red {font: bold 2.0em Helvetica, sans-serif; color: red}
w {text-decoration: underline;}
x {font-weight: bold;}
todo {font: bold 11pt "Helvetica", sans-serif; padding: 1px 2px; margin: 2px;
      border: thin solid red;}
todo:before {content: "TODO: "; color: red;}
pre {margin-left: 4em; font: bold 11pt "Courier", serif;}
inline {font: bold 11pt "Courier", serif;}
prob {font: italic 11pt "Helvetica", sans-serif; background-color: #FFFFFF;}
sol {font: 12pt "Times New Roman", serif;}
sol:before {content: "Possible solution:  "; color:  #0099DD; }
div#ctr {font: bold 1.5em Helvetica, sans-serif; color: #0099DD;
    text-align: center;}
warn {font: italic 12pt "Times New Roman", serif; color: red}
</style>
</head>

<body>
<hr>
<p>
<a href="http://www.quantumbioinc.com" target="QBHomePage"> <object height=51
  align=left
  data="images/qb_logo.png" type=image/png title="QuantumBio Homepage">
QuantumBio Homepage
</object> </a>
<a href="http://www.quantumbioinc.com"target="QBHomePage"> <object height=51
  align=right
  data="images/qb_logo.png" type=image/png title="QuantumBio Homepage">
QuantumBio Homepage
</object> </a>
<h1>MOEDivCon Integration Suite</h1>
<hr>
<h2><a name="ToC"></a>Table of Contents</h2>
<ol id=tocl>
  <li><a href="#ToC">Table of Contents</a>
  <li><a href="#intro">Introduction</a>
  <li><a href="#install">Installation</a>
  <li><a href="#start">Getting Started</a>
  <ol id=tocl>
    <li><a href="#start-onetoone">One-to-One Mode</a>
    <li><a href="#start-onetomany">One-to-Many Mode</a>
    <li><a href="#start-manytomany">Many-to-Many Mode</a>
  </ol>
  <li><a href='#core'>MOEDivCon Core</a>
    <ol id=tocl>
      <li><a href="#core-intro">Introduction</a>
      <li><a href="#core-test">Molecule Tests</a>
      <ol id=tocl>
        <li><a href="#core-test-lp">Lone Pairs</a>
        <li><a href="#core-test-overlap">Atom Overlap</a>
        <li><a href="#core-test-hamcompat">Hamiltonian Compatibility</a>
        <li><a href="#core-test-h">Missing/Extra H</a>
        <li><a href="#core-test-mm">Undefined MM Atom Type</a>
      </ol>
      <li><a href="#core-charge">Charges</a>
      <ol id=tocl>
        <li><a href="#core-charge-status">Status</a>
        <li><a href="#core-charge-onetomany">One-to-Many Mode</a>
        <li><a href="#core-charge-manytomany">Many-to-Many Mode</a>
        <li><a href="#core-charge-editing">Editing</a>
      </ol>
      <li><a href="#core-editmol">Editing Molecules</a>
      <li><a href="#core-load">Load, Create, and Update</a>
      <li><a href="#core-clean">Log Files and Postexecution Cleanup</a>
      <li><a href="#core-parallel">Parallel Execution</a>
      <ol  id=tocl>
        <li><a href="#core-parallel-mode">Parallel Mapping Mode</a>
        <li><a href="#core-parallel-moebatch">Running Moebatch/MPI</a>
      </ol>
      <li><a href="#core-options">Options</a>
      <ol id=tocl>
        <li><a href="#core-options-outdb">Output MOE Database</a>
        <li><a href="#core-options-workdir">Working Directory</a>
        <li><a href="#core-options-ham">Hamiltonian</a>
        <li><a href="#core-options-scaling">Scaling</a>
      </ol>
    </ol>
  <li><a href="#nmr">MOEDivCon NMRScore</a>
    <ol id=tocl>
      <li><a href="#nmr-intro">Introduction</a>
      <li><a href="#nmr-workflow">Workflow</a>
      <li><a href="#nmr-cut">Receptor Cut</a>
      <li><a href="#nmr-csp">Chemical Shift Atoms</a>
      <ol id=tocl>
        <li><a href="#nmr-csp-type">Atom Type</a>
        <li><a href="#nmr-csp-exp">Experimental Data</a>
      </ol>
      <li><a href="#nmr-options">Options</a>
      <ol id=tocl>
        <li><a href="#nmr-options-opt">Ligand Optimization</a>
        <li><a href="#nmr-options-guess">Guess</a>
      </ol>
<!--      <li><a href="#nmr-results">Results</a>  -->
    </ol>
  <li><a href="#qms">MOEDivCon QMScore</a>
    <ol id=tocl>
      <li><a href="#qms-intro">Introduction</a>
      <li><a href="#qms-options">Options</a>
      <ol id=tocl>
        <li><a href="#qms-options-opt">Ligand Optimization</a>
        <li><a href="#qms-options-shift">Shift</a>
        <li><a href="#qms-options-freq">Frequency</a>
        <li><a href="#qms-options-pwd">PWD</a>
      </ol>
      <li><a href="#qms-results">Results</a>
    </ol>
  <li><a href="#pwd">MOEDivCon QB-PWD Analysis</a>
    <ol id=tocl>
      <li><a href="#pwd-intro">Introduction</a>
      <li><a href="#pwd-workflow">Workflow</a>
      <li><a href="#pwd-align">Protein Alignment</a>
      <li><a href="#pwd-analysis">Analysis</a>
      <ol id=tocl>
        <li><a href="#pwd-analysis-view">View</a>
        <li><a href="#pwd-analysis-save">Save</a>
      </ol>
      <li><a href="#pwd-options">Options</a>
      <ol id=tocl>
        <li><a href="#pwd-options-title">Title</a>
        <li><a href="#pwd-options-scorefun">Scoring Functions</a>
        <li><a href="#pwd-options-residues">Residues</a>
        <li><a href="#pwd-options-pls">PLS</a>
      </ol>
    </ol>
  <li><a href="#heatmap">MOEDivCon Heat Map</a>
  <ol id=tocl>
    <li><a href="#heatmap-selection">Selection</a>
    <li><a href="#heatmap-saveprint">Save and Print</a>
    <li><a href="#heatmap-animation">Animation</a>
    <li><a href="#heatmap-setup">Configuration</a>
      <ol id=tocl>
        <li><a href="#heatmap-setup-cutoff">Cutoff</a>
        <li><a href="#heatmap-setup-numcolors">Number of Colors</a>
        <li><a href="#heatmap-setup-mono">Monochromatic</a>
        <li><a href="#heatmap-setup-numticks">Number of Ticks</a>
      </ol>
  </ol>

  <li><a href="#trouble">Troubleshooting</a>
  <li><a href="#contact">Support Contact Information</a></ol>

<h2><a name="intro"></a>Introduction</h2>
<p>
QuantumBio offers the DivCon Discovery Suite product line, providing
solutions to achieve high accuracy, performance, and versatility for chemical
characterization in drug discovery and development. The DivCon Discovery Suite
is built on cutting-edge technology that utilizes precise quantum mechanical
algorithms in a user-friendly format, providing the opportunity for faster
results and reduced costs.
<p>
QuantumBio offers the following lead optimization capabilities for your drug
discovery investment:
<ul>
<li> Use QM based Scoring Function to predict protein-ligand binding affinities
     on a routine basis.
<li> Build computational expensive QM descriptors for QSAR & ADME/Tox Models.
<li> Study protein-ligand complexes (even those containing metals).
<li> Predict ligand-induced NMR (Nuclear Magnetic Resonance) Chemical Shift
     Perturbation on protein.
<li> Study large biomolecules of 500 or 15,000 atoms with QM.
<li> Understand the PairWise Energy Decomposition of protein-ligand complex
     interaction.
<li> Leverage the accuracy and speed of divide & conquer linear-scaling QM.
</ul>

<h2><a name="install"></a>Installation</h2>
<p>
The MOEDivCon Integration is part of the QuantumBio Discovery Suite 
Distribution, which is packaged in a
<inline>tar.gz</inline> file that uses the following naming convention
<inline>quantumbio-&lt;platform&gt;-&lt;version&gt;.tar.gz</inline>.  Move
the file where you wish to install the QuantumBio software (e.g.,
<inline>/usr/local</inline>), then uncompress and untar the file.
<pre>
&gt; tar  -zxvf  quantumbio-&lt;platform&gt;-&lt;version&gt;.tar.gz
</pre>
We will
use <inline>$QBHOME</inline> to refer to the path where the package is
installed.  For instance, if you install linux version 5.0.0.1 in
the <inline>/usr/local</inline> directory,
<inline>$QBHOME</inline> will
refer to <inline>/usr/local/quantumbio-linux-5.0.0.1</inline>.

<p><a name="install-menu"></a>
The final steps of the installation set up your environment to run MOEDivCon.
First, you need to copy the MOEDivCon menu file to your home directory.  If
you already have a custom MOE menu, you need to concatenate the files.
For example, either
<pre>
&gt; cp  ${QBHOME}/moe-menus  ${HOME}/moe-menus
</pre>
or
<pre>
&gt; cat  ${YOURMENUDIR}/moe-menus  ${QBHOME}/moe-menus  >  ${HOME}/moe-menus
</pre>

<p><a name="install-svl"></a>
In order for
MOE to locate the MOEDivCon SVL library, you must create a link to the
library in a place where MOE can find it.  If you already have a directory
of SVL scripts, link the MOEDivCon SVL libarary as a subdirectory.
For example:
<pre>
&gt; ln -s  ${QBHOME}/svl  ${MOE_SVL_PATH}/qb_lib.svl
</pre>
Otherwise, create an SVL directory in your home directory and link the
MOEDivCon SVL library under it.  For example:
<pre>
&gt; mkdir  ${HOME}/svl<br>
&gt; ln -s  ${QBHOME}/svl  ${HOME}/svl/qb_lib.svl<br>
</pre>

<p><a name="install-path"></a>
Before you can run MOEDivCon Suite using MOE or moebatch, 
you need to make sure that
the MOEDivCon executables are in your path and the MOEDivCon SVL library is in
your MOE run path.  For example, you should add to your .bashrc file:
<pre>
export  PATH=${QBHOME}/bin:${PATH}<br>
if [ -z  $MOE_SVL_RUNPATH ] ; then<br>
  export  MOE_SVL_RUNPATH="${QBHOME}/svl"<br>
else<br>
  export  MOE_SVL_RUNPATH="${QBHOME}/svl:$MOE_SVL_RUNPATH"<br>
fi<br>
</pre>
or add to your .cshrc file:
<pre>
setenv  PATH  ${QBHOME}/bin:${PATH}<br>
if ( $?MOE_SVL_RUNPATH ) then<br>
  setenv  MOE_SVL_RUNPATH  ${QBHOME}/svl:$MOE_SVL_RUNPATH<br>
else<br>
  setenv  MOE_SVL_RUNPATH  ${QBHOME}/svl<br>endif<br>
</pre>

<p><a name="install-license"></a>
Finally, MOEDivCon is distributed with an evaluation license that will
eventually
expire. Contact sales@quantumbioinc.com to purchase a license file.
Once you have received a license file
from QuantumBio, you can install the file with a simple move statement.
<pre>
&gt; mv  license-file  $QBHOME/etc/license
</pre>
If you are having difficulties with the installation, see the
<a href="#contact">Support Contact Information</a> section.

<h2><a name="start"></a>Getting Started</h2>
<p>
To verify that the installation was successful, let's work through a few
simple examples of QMScore.  If you run into any problems, 
see the <a href="#trouble">Troubleshooting</a> section.
<p>
The workflow of MOEDivCon QMScore involves six steps:
<ol>
  <li> Prepare the receptors and ligands/poses involved in the calculation.
  <li> Closely examine all error and warning produced by the MOEDivCon tests
       to ensure that the receptor(s) and ligand(s) are not problematic.
  <li> Verify the formal charges calculated by MOE.
  <li> Set up the run parameters to QMScore.
  <li> Execute QMScore either on a local machine or in parallel on a cluster.
  <li> Analyze the QMScore results.
</ol>

<h3><a name="start-onetoone"></a>One-to-One Mode</h3>
<p>
First, start MOE and open
<inline>${QBHOME}/examples/simplecomplex.moe</inline>.  Then,
select the MOE menu
<w>Compute</w>&rarr;<w>Simulations</w>&rarr;<w>QuantumBio</w>&rarr;<w>QMScore</w>.
<p>
<object id="MOEwithMenu" title="Moe Window with QuantumBio Menu" width=956
data="images/mainwithmenu.png" type=image/png>
Moe Window with QuantumBio Menu
</object>
<p>
The QMScore panel should appear with the default values.
<p>
<object id="QMScoreDefaultDialog" title="QMScore Panel"
data="images/qmscoredefault.png" type=image/png>
QMScore Panel
</object>
<p>
Unselect <w>CWD</w> and ensure that the <w>Working Dir</w> is
<inline>/tmp</inline> (files for storing intermediate calculations and logs
 will be written here).  Press the <w>Run</w> button.  If all goes
well, QMScore will execute using the receptor and ligand complex shown in
the MOE Window.  After several seconds, the MOE database Viewer should open
the QMScore database showing the results.  

Note that DivCon creates an <inline>HDF5</inline> file with the same name 
(extension <inline>.h5</inline>) and in the same directory as the QMScore 
database, but the temporary
files in <inline>/tmp</inline> are deleted when DivCon returns successfully.

<p>
<object id="QMScoreDatabase" title="QMScore Database" width=956 
data="images/qmscoredatabase.png" type=image/png>
QMScore Database
</object>
<p>
In this simple example, you were able to perform all four workflow steps
in one session within the MOE Window, probably not something you would wish
to do with a large protein receptor for which the execution time may take
several hours.
<h3><a name="start-onetomany"></a>One-to-Many Mode</h3>
<p>
The real power of the MOEDivCon Integration is the ability to use moebatch
and MPI to execute tasks in parallel on a cluster
and collate the results in a MOE database.  In this case, start MOE and open
<inline>${QBHOME}/examples/simplecomplex.moe</inline> and select the MOE menu
<w>Compute</w>&rarr;<w>Simulations</w>&rarr;<w>QuantumBio</w>&rarr;<w>QMScore</w>.
Select the <w>One-to-Many</w> mapping mode.  Browse for the poses database
and select <inline>${QBHOME}/examples/simpleposes.mdb</inline>.  In the
<w>Charges</w> section, select <w>All Poses</w> since the database contains 
different poses of the same ligand.  Your settings should look like:
<p>
<object id="QMScoreOptionsOneToMany" title="QMScore One-to-Many Options"
data="images/qmscoreoptionsonetomany.png" type=image/png>
QMScore One-to-Many Options
</object>
<p>
Press <w>Create</w> and verify that the charges and other data for the poses
are okay (the <w>Status</w> field is empty).  Close the MOE Database Viewer
and click the <w>Run</w> button.
The jobs will be executed sequentially in the current MOE session (not
recommended for large proteins).
MOEDivCon will print out progress messages.  When all
four jobs are completed, open <inline>qmscore.mdb</inline> in the MOE
Database Viewer.  You should see results similar to these:
<p>
<object id="QMScoreDatabaseOneToMany" title="QMScore One-to-Many Database" width=956 
data="images/qmscoredatabaseonetomany.png" type=image/png>
QMScore Database
</object>

See the <a href="#core-parallel">Parallel Execution</a> section for
more information on executing moebatch in parallel on a cluster.
<h3><a name="start-manytomany"></a>Many-to-Many Mode</h3>
<p>
In Many-to-Many Mode, MOEDivCon doesn't use the molecules in the MOE Window,
so after starting MOE, selecting <w>QMScore</w> from the menu system,
and selecting
<w>Many-to-Many</w> mapping mode, and you can proceed directly to browsing
for the receptor database (<inline>${QBHOME}/examples/simplerecs.mdb</inline>) 
and ligand database (<inline>${QBHOME}/examples/simpleligands.mdb</inline>).
Also change the working directory to <inline>~</inline> or some other network
mounted directory accessable by all of the nodes on your computing cluster.
Your settings should look like:
<p>
<object id="QMScoreOptionsManyToMany" title="QMScore Many-to-Many Options"
data="images/qmscoreoptionsmanytomany.png" type=image/png>
QMScore One-to-Many Options
</object>
<p>
Note that the <w>Charges</w> section is inactivated; the charges are
calculated as the receptors and ligands are copied to the QMScore database.
Press <w>Create</w> and verify that the charges and other data for both 
the receptors and ligands are okay (the
<w>Status</w> field is empty).  Close the MOE Database Viewer, copy
<inline>mmoedivcon_qmsmpi_bash.pbs</inline> from
 <inline>${QBHOME}/scripts/</inline> to the directory 
where <inline>qmscore.mdb</inline>
is located, change the number of processors (if necessary) 
and submit it to your PBS queue:
<pre>
&gt; qsub moedivcon_qmsmpi_bash.pbs
</pre>
Analagous steps are followed for other cluster management systems.  When all
twelve jobs are completed, open <inline>qmscore.mdb</inline> in the MOE
Database Viewer.  You should see results similar to these:
<p>
<object id="QMScoreDatabaseManyToMany" title="QMScore Many-to-Many Database" width=956 
data="images/qmscoredatabasemanytomany.png" type=image/png>
QMScore Database
</object>
See the <a href="#core-parallel">Parallel Execution</a> section for
more information on executing jobs in parallel on a cluster.


<h2><a name="core"></a>MOEDivCon Core</h2>
<h3><a name="core-intro"></a>Introduction</h3>
<p>
MOEDivCon Core encompasses the GUI form and functionality that is shared by all
of the Score Calculation panels in the MOEDivCon Integration Suite.  It includes
a set of molecular tests to capture common structural problems in the protein 
receptor(s) and ligand(s) before they are sent to DivCon for evalutation.  It
also includes a verification that the charges calculated by MOE are compatible
with DivCon.  It includes basic functionality such as saving and loading
MOEDivCon Score database, and options common to all scoring calculations.
Finally, it includes the support for setting up jobs to run in batch mode and
the ability to execute jobs in parallel on either a workstation or computing
cluster.

<h3><a name="core-test"></a>Molecule Tests</h3>
<p>
MOEDivCon is able to perform several tests on the receptor(s) and ligand(s) to
determine if there are possible problems with the molecular structures.  The 
"Error" tests check whether there are problems that will definitely cause DivCon
to fail and the "Warning" tests check for other possible problems that may or
may not cause DivCon to fail (or not converge), but may cause questionable
calculation results.
<p>
Because the tests can be computationally intensive on large proteins
and sometimes report false positives,
you are able to select which tests are performed during setup.
You should turn off the tests only if you are sure the receptor(s) and ligand(s)
have been cleaned up and are correctly parsed by MOE.
<p>
The "Error" tests are always performed on the receptor and ligand in the MOE
Window when you click <w>Run</w> or <w>Create</w> in <w>One-to-One</w> or
<w>One-to-Many</w> mode.  If any problems  that will cause DivCon to fail are
detected, you are asked whether you wish to continue.  The selected tests are
also performed on any receptors and/or ligands/poses as they are copied into
the MOEDivCon Score database in <w>One-to-Many</w> or <w>Many-to-Many</w> mode.  The
results of the tests are stored in the <w>Status</w> field of the 
MOEDivCon Score database so that you may verify the test results.
<p>
The "Error" tests are always performed by MOEDivCon before passing the
receptor and ligand to DivCon for calculation.  If any of the "Error" tests
fail, the calculation will be skipped.  The "Error" and selected "Warning"
test results are stored in the <w>Status</w> field of the MOEDivCon Score 
database.
<p>
For each atom that is detected to be problematic, MOEDivCon prints information
that makes it easy to locate the atom using MOE's Atom Manager.  The 
statement has the following format:
<pre>
AtomIndex: AtomName in ResidueIndex: ResidueName ResidueUID
</pre>
where the AtomIndex is MOE's index in the system, not the residue.  
The ResidueIndex is MOE's index of the residue in the system, 
and ResidueUID is original ID field from
the pdb or mol2 file, say.  Below is an example error and warning tests report
(for 1KZK from the Astex Set).
<p>
<object id="QMScoreErrorReport" title="QMScore Error and Warning Tests Report"
data="images/qmscoreerrorreport.png" type=image/png>
QMScore Error and Warning Tests Report
</object>
<p>

<h4><a name="core-test-lp"></a>Lone Pairs</h4>
<p>
This is an "Error" test.
DivCon doesn't recognize MOE's <inline>LP</inline> (lone pair) element name and
will immediately abort the calculation of a molecule that contains one.  The
most likely reason for a lone pair appearing in a molecule is that MOE had
problems parsing a PDB file, in particular the Atom Name field.

<h4><a name="core-test-overlap"></a>Atom Overlap</h4>
<p>
This is an "Error" test.
Not unexpectedly, DivCon has serious convergence problems if there are unbonded
atoms with overlapping orbitals.  This test checks for intramolecular
atom overlap; that is, whether any two atoms in the ligand are too close to 
each other, or any two atoms in the receptor collection, which may include 
solvents, cofactors, etc.  This test detects structural problems that may be
difficult to see with a 3D molecular viewer.  This test also checks for
intermolecular atom overlap; that is between receptor and ligand atoms.  At
the moment, the test uses a simple heuristic that will be improved with time.

<h4><a name="core-test-hamcompat"></a>Hamiltonian Compatibility</h4>
<p>
This is an "Error" test.
If there exists an element in either the receptor or ligand for which the
selected Hamiltonian has not been parameterized, DivCon is going to eventually
abort its calculation.  This test captures that failure event before performing
any time-consuming calculations.
<p>
The elements parameterized for each Hamiltonian are listed below:

<ul>NMRScore:
  <li>AM1:  H, Li, Be, B, C, N, O, F, Na, Al, Si, P,
            S, Cl, K, Ca, Zn, Ge, Br, I, Hg
  <li>MNDO: H, C, N, O, P, S, Cl, B, F, Zn, Br
</ul>

<ul>QMScore:
  <li>AM1: H, Li, Be, B, C, N, O, F, Na, Al, Si, P,
            S, Cl, K, Ca, Zn, Ge, Br, I, Hg
  <li>MNDO:  H, Li, Be, B, C, N, O, F, Na, Al, Si, P,
            S, Cl, Zn, Ge, Br, Sn, I, Hg, Pb
  <li>MNDO/d:  H, Li, Be, B, C, N, O, F, Na, Mg, Al, Si,
            P, S, Cl, Zn, Br, Cd, I, Hg
  <li>PDDG-PM3:   H, C, N, O, F, S, Cl, Br, I
  <li>PM3:  H, Be, C, N, O, F, Na, Mg, Al, Si, P, S,
            Cl, K, Ca, Zn, Ga, Ge, Ar, Se, Br, Cd, In,
            Sn, Sb, Te, I, Hg, Tl, Pb, Bi
  <li>PM6:  H, Li, Be, B, C, N, O, F, Na, Mg, Al, Si,
            P, S, Cl, K, Fe, Zn, Br, Cd, I, Hg
</ul>


<h4><a name="core-test-h"></a>Missing/Extra H</h4>
<p>
This is a "Warning" test.
The most common reasons for a missing H are truncated residues in proteins
and misinterpreted bond orders in ligands.  The most common reason for an
extra H in proteins are protonated carbonyl oxygens (C=O&rarr;C=OH).   Other
problems are caused by incorrect charges, metal coordinations, etc.
<p>
This also tests for the presence of positively charged carbon (C<sup>+</sup>),
often caused by incorrect bond orders in the guanidinium group in arginine and
the imidazole ring in histidine.
<p>
This test can also generate false positives.  MOEDivCon already filters out
the most common source of false positives:  terminal amines in proteins
treated by MOE as ammonion cations. 
<p>
It is recommended that you verify the correctness of the molecular structure
if you receive this warning.

<h4><a name="core-test-mm"></a>Undefined MM Atom Type</h4>
<p>
This is a "Warning" test.  If MOE was unable to determine the molecular
mechanics atom type, it's very likely that there is a serious problem with
the structure.  If an atom has a problem detected by the other tests, it 
is often listed under this test as well.  It also detects problems not
caught by the other tests such as truncated residues, incorrect charges on
metals, incorrect bond orders on sulphates, sulfoxides, sulfonyls, 
phosphates, and orthovanadate ions, metal coordinations problems, etc.
<p>
It is recommended that you verify the correctness of the molecular structure
if you receive this warning.

<h3><a name="core-charge"></a>Charges</h3>
<p>
After correct coordinates, the second most important molecular descriptor on
which DivCon depends is the total charge. DivCon calculations
rely on an accurate total charge in order to properly characterize a molecule.
Improper or incorrect charges can result in poor energies, convergence problems,
questionable geometries, and incorrect partial charges. The total charge can
also yield a system with an odd number of electrons. Such systems are not only
highly unusual in biological problems, they are also incompatible the with
underlying Restricted Hatree Fock (RHF) algorithm within DivCon. 
MOE is capable of determining the charge based upon bond order, protonation,
and analogous characteristics. In the event that the charge is incorrectly 
determined, it is suggested that the user edit the structure using MOE's
molecule Builder. Please see the MOE documentation for more information. 
<p>
When you open any of the MOEDivCon Score panels, MOE calculates the total
charge of the
receptor and ligand in the MOE window, and displays them in the
<w>Charges</w> section.  You can press the <w>Calculate</w> button to
recalculate the charges.

<h4><a name="core-charge-status"></a>Status</h4>
<p>
The colored cirles next to the charges reflect the status of the charge
calculation:
<table border="1">
<tr><th class="white">&#X25CF;<td>Uncalculated or user-entered charge.
<tr><th class="green">&#X25CF;<td>Charge calculated by MOE.
<tr><th class="red">&#X25CF;<td>Problematic charge for DivCon.
</table>
<p>
Note that the green circle doesn't mean that the charge is correct, only that
it has been calculated.  As with any other formal charge algorithm, MOE 
doesn't always get the total charge right.  You can edit the
charges calculated by MOE if you think they are incorrect or they are marked
as problematic for DivCon.  If you modify the charges in the 
MOEDivCon Score panel,
MOEDivCon will alert you if a charge you entered may cause problems with the
DivCon calculations when you press <w>Run</w> or <w>Create</w>.
<p>
<object id="QMScoreBadCharge" title="QMScore Charge Error"
data="images/qmscorebadcharge.png" type=image/png>
MOEDivCon Score Charge Error
</object>

<h4><a name="core-charge-onetomany"></a>One-to-Many</h4>
<p>
When in <w>One-to-Many</w> mode, you can choose
to use the charge specified for the ligand as the charge for all the poses
by selecting <w>All Poses</w>.  Otherwise, the charge is calculated for
each pose as it is copied from the Ligand/Pose database to the
MOEDivCon Score database.  In either case, when the MOEDivCon Score database
is opened with
the MOE Database Viewer for verification.  If a calculated charge is
problematic for DivCon, it is noted in the <w>Status</w> column (double 
click cell to see entire message).  DivCon calculations for molecules with
problematic charges are skipped.

<h4><a name="core-charge-manytomany"></a>Many-to-Many</h4>
<p>
When in <w>Many-to-Many</w> mode, the <w>Charges</w>
section is disabled.  The charges for the receptors and ligands are calculated
as they are copied from their respective databases into the MOEDivCon Score
database.  Again, you can verify and edit the charges using the MOE Database
Viewer.  DivCon calculations for molecules with problematic charges are skipped.

<h4><a name="core-charge-editing"></a>Editing</h4>
<p>
You can edit the charges stored in the MOEDivCon Score database by double
clicking
on the cell and edit the charge in the edit bar at the top of the window.
Afterwards, you can press the <w>Check</w> button in the <w>Charges</w>
section of the panel to ask MOEDivCon to recheck the charges in the 
MOEDivCon Score
database.  If a charge is still problematic, a warning message appears in
the <w>Status</w> column.  Otherwise, an okay message is written.
<p>
Before passing the run parameters to DivCon
during execution, MOEDivCon checks again whether the charges are
going to be problematic.  If so, it skips the calculation and places a
message in the <w>Status</w> column.
<p>
<warn>Warning:  When you edit a cell in a MOE database, the new value is
directly written to disk and the previous value is lost.</warn>

<h3><a name="core-editmol"></a>Editing Molecules</h3>
<p>
If you do encounter a problematic molecule, you may wish to edit it to fix the
problem.  See the MOE documentation for a detailed description of how to use 
MOE's tools for editing and optimizing molecules, but here's a few additional
pointers relevant for the MOEDivCon Integration Suit.
<p>
<ul>
  <li>Close all MOEDivCon panels before editing molecules.
  <li>If you open an existing MOEDivCon Score database and edit some of the
    <w>Receptor</w> or <w>Ligand/Pose</w> fields directly using
    <w>Get from MOE</w> and <w>Send to MOE</w> options from the 
    <w>Molecule</w> popup menu, then you need to rename the receptors and
    ligands including the MOEDivCon specific tags:  "rec#" and "lig#" (or
    "rec#=<em>&lt;name&gt;</em>" and "lig#=<em>&lt;name&gt;</em>"
    if the molecules also have a user supplied name). To do so, right click
    on the appropriate molecule cell and select <w>Set Name</w> from the popup
    menu.
  <li>It is suggested that you also save any edited molecules in their
    original file format (mol2, pdb, etc.), so that, if necessary, the
    receptor and/or ligand/pose databases can be regenerated with the
    correct structures.
  <li>If you modify a MOEDivcon Score database after it has been submitted
    for parallel execution on a computing cluster to correct problematic
    molecules, you can resubmit it to the cluster queue and the jobs marked
    successful in the database will be skipped.
</ul>

<h3><a name="core-load"></a>Load, Create, and Update</h3>
<p>
You can load an existing MOEDivCon Score database into the MOEDivCon Score
panel by pressing the <w>Load</w> button.  A file selection box will open
allowing you to browse for the MOEDivCon Score database.  When you select 
a MOE database, it is checked to make
sure it is a MOEDivCon Score database.  The receptor/ligand complex stored in
the database is then loaded into the MOE window (if the mode of the database
is <w>Many-to-Many</w>, then the first receptor and ligand are loaded),
the MOEDivCon Score panel is
updated with the settings stored in the database, and then the database itself
is opened using the MOE Database Viewer.
<p>
If you load an existing MOEDivCon Score database and change the options and/or
charges in the panel, you can then press <w>Update</w> to save the new options
and/or charges in the output database.  If you also edit the receptor or ligand
in the MOE Window using MOE's Builder, then press <w>Create</w>, and the 
updated options, molecules and charges will be saved in the output database.
<p>
If you load an existing MOEDivCon Score database and only edit some of the
<w>Rec Chg</w> or <w>Lig Chg</w> fields, or edit the <w>Receptor</w> or 
<w>Ligand/Pose</w> field directly using <w>Get from MOE</w> 
and <w>Send to MOE</w> options from the <w>Molecule</w> popup menu, then
you shouldn't press the MOEDivCon Score <w>Create</w> button.  This would 
regenerate the original molecules
and charges
calculated from the molecules in the MOE Window, and/or those stored in the
recepter and pose database specified in MOEDivCon Score panel.  When you edit
an entry in a MOE database, it is written directly to disk.  See the
<a href="#core-editmol">editing molecule</a> and
<a href="#core-charge-editing">editing charge</a> sections for more information.
<p>
<warn>Warning:  When you load a database, the current settings in the
MOEDivCon Score panel and the molecules in the MOE Window are replaced
and may be lost if not saved beforehand.</warn>

<h3><a name="core-clean"></a>Log Files and Postexecution Cleanup</h3>
<p>
If a DivCon job runs successfully, MOEDivCon automatically deletes
temporary HDF5 (.h5 extension) files in the working directory.  However,
if an error is reported the files are not deleted and the location of the
DivCon log file is written to the <w>Status</w> column of the MOEDivCon Score
database along with the error message.  The location has the format:
<pre>
        hostname:/path/to/working/directory/divcon.log
</pre>
The hostname is useful if the job ran on a cluster and you are not sure on
which particular machine.  This allows you to examine the DivCon output and
logs if you need more information about why the error occurred.  It also makes
it easier for you to find the working directory in order to delete the
temporary files post-mortem.

<h3><a name="core-parallel"></a>Parallel Execution</h3>
<p>
As described in the <a href="#start">Getting Started</a> section, the workflow
of MOEDivCon QMSore proceeds in six steps:  preparation, testing, verification,
set up, execution, and analysis.  The prep, testing, verification, set up, and
analysis steps are performed using MOE on your local workstation or laptop.
The third step, execution, is best left to a computational cluster, whether 
you are scoring one complex or hundreds.
<p>
Before getting into the details of executing moebatch on a cluster, we'll
explain how to setup MOEDivCon QMScore to run many jobs in parallel.  
Analogous steps are performed to set up MOEDivCon NMRScore jobs.
<p>
Note:  the current working directory must be mounted on all computing cluster
nodes, unless all processors used for the parallel run are on the same node.
For instance, <inline>/tmp</inline> is locally mounted on each node and should
not be used for the current working directory of a parallel run.

<h4><a name="core-parallel-mode"></a>Mapping Mode</h4>
<p>
The mapping mode determines how receptors are to be mapped to ligands (poses).

<h5><a name="core-parallel-mode-1to1"></a>One Receptor--One Ligand</h5>
<p>
In <w>One-to-One</w> batch mode, the set up is the same as in the
<a href="#start">Getting Started</a> section.  Open the receptor
and ligand files into the MOE Window and perform any necessary preparatory,
testing and verification steps.
Next, open the QMScore panel and select your options, and then choose
<w>Create</w> rather than <w>Run</w>.  MOEDivCon will store your receptor,
ligand, and
options in the QMScore database, and then open the MOE Database Viewer so you
can verify the ligand/pose data.  
See the <a href="#core-parallel-moebatch">Running Moebatch/MPI</a> section
to learn how to execute the job.

<h5><a name="core-parallel-mode-1toM"></a>One Receptor--Many Ligand Poses</h5>
<p>
In <w>One-to-Many</w> parallel mode, MOEDivCon scores many ligand poses
generated by your favorite docking program against the same receptor.  The set
up is similar to <a href="#core-parallel-mode-1to1"><w>One-to-One</w></a> mode
in that MOEDivCon expects the receptor/ligand complex to be in the MOE Window.
The difference is that you need to create a MOE database to store the ligand
poses beforehand (see the MOE documentation if you are not familiar with
this process).
<p>
Open the receptor and ligand files into the MOE Window, open the QMScore
panel and choose your
options.  When you select the <w>One-to-Many</w> mode, you will be prompted
to supply a ligand/pose database.
<p>
<object id="QMScoreOneToMany" title="QMScore One-to-Many Mode"
data="images/qmscoreonetomany.png" type=image/png >
QMScore One-to-Many Mode
</object>
<p>
You can choose to view all poses simultaneously in the MOE Window by selecting
<w>All Poses</w> in the Mapping Mode section.
You can also choose
to use the charge specified for the ligand as the charge for all the poses
by selecting <w>All Poses</w> in the Charge section.  
See the <a href="#core-charge-onetomany">Charge</a>
section to learn more details about how MOEDivCon handles charges.
<p>
When you choose <w>Create</w>,  MOEDivCon will store your receptor, ligand, and
options in the QMScore database, and it will copy the ligand/pose database into
the QMScore database including any data fields (e.g., results from other scoring
algorithms).  The MOE Database Viewer will then open the QMScore database so
you can verify the ligand/pose data, especially the charge values and status.
See the <a href="#core-parallel-moebatch">Running Moebatch/MPI</a>
section to learn how to execute the job.
<p>
<h5><a name="core-parallel-mode-MtoM"></a>Many Receptors--Many Ligands</h5>
<p>
In the <w>Many-to-Many</w> parallel mode, MOEDivCon scores many receptors,
each against its own ligand.  In this case, MOEDivCon gathers the molecular
information from the two databases you need to supply.  Additionally, it
is up to the user to assure that the receptors and ligands are stored in the
correct order in the MOE databases.
<p>
<object id="QMScoreManyToMany" title="QMScore Many-to-Many Mode"
data="images/qmscoremanytomany.png" type=image/png>
QMScore Many-to-Many Mode
</object>
<p>
When you choose <w>Create</w>,  MOEDivCon will store your options in the
QMScore database, and it will copy the receptor and ligand databases, row
by row, into the QMScore database including any data fields.  The MOE
Database Viewer will then open the QMScore database so you can verify the
receptor and ligand data, especially the charge values and status.
See the <a href="#core-parallel-moebatch">Running Moebatch/MPI</a> section
to learn how to execute the job.  See the
<a href="#core-charge-manytomany">Charge</a>
section to learn more details about how MOEDivCon handles charges.
<p>
<warn>Warning:  If the receptor and pose databases have fields&ndash;other than
the molecule field&ndash;with the same name, then the pose data will overwrite
the receptor data for those fields.</warn>


<h4><a name="core-parallel-moebatch"></a>Running Moebatch/MPI</h4>
<p>
After you close the database, you can execute MOEDivCon QMScore by submitting
a script to a cluster queue (analogous steps are performed to execute MOEDivCon
NMRScore.).  Example
scripts for PBS and SGE are located in <inline>${QBHOME}/scripts</inline>.
For example, below is the script for submitting a parallel job to PBS
using bash (<inline>${QBHOME}/scripts/moedivcon_qmsmpi_bash.pbs</inline>):
<pre>
#!/bin/bash
#PBS -S /bin/bash
#PBS -l nodes=2:ppn=7
#PBS -l mem=2000mb
#PBS -j oe

hostname
date
cat $PBS_NODEFILE

# Change the path and name for the MOE database.
USER_DATADB="$PBS_O_WORKDIR/qmscore.mdb"

moebatch -exec "run ['qmsmpi.svl', ['$USER_DATADB', '$PBS_JOBID']]" -exit
</pre>
The MPI implementation of MOEDivCon require one process for moebatch and a 
separate processor for each worker that performs QMScore computations.
The above script requests 14 (2 * 7) proccessors to use in the parallel job,
so thirteen would be available to process the receptor/pose complexes in the 
QMScore database.  PBS generates the list of nodes in the 
<inline>PBS_NODEFILE</inline> that is used by MPI.  Note that only one MOE 
license for moebatch is required.
<p>
The first argument to <inline>qmsmpi.svl</inline>, 
<inline>$USER_DATADB</inline>, is the path to your QMScore database. The second
argument, which is optional, is the name of the subdirectory to use to store
temporary files in the working directory; in this case, 
<inline>$PBS_JOBID</inline>.  A unique subdirectory name ensures that
different parallel jobs don't mix temporary files.
<p>
Each worker process creates a single log file in the <inline>log</inline>
subdirectory of the working directory for all of the jobs it executes.
Each worker process creates a separate <inline>HDF5</inline> file in the 
working directory for each job that it executes.  These files are automatically
deleted after a successful execution.
<p>
In addition to storing the scores in the QMScore MOE database, the QMScore
application creates an <inline>HDF5</inline> file in the same location and with
the same name as the QMScore database (<inline>.h5</inline> extension).  This
file contains additional data (for instance, PWD) that the application uses for
post-execution analysis.  Note that if you resubmit a QMScore database, those
rows that have "Success" in the <w>Status</w> field are skipped.
<p>
An alternate setup, especially for users who prefer to use the <w>CWD</w>
option for the working directory would be:

<pre>
#!/bin/bash
#PBS -S /bin/bash
#PBS -l nodes=2:ppn=7
#PBS -l mem=2000mb
#PBS -j oe

hostname
date
cat $PBS_NODEFILE

mkdir $PBS_JOBID; cd $PBS_JOBID

# Change the path and name for the MOE database.
USER_DATADB="qmscore.mdb"

moebatch -exec "run ['qmsmpi.svl', ['$USER_DATADB']]" -exit
</pre>

<h3><a name="core-options"></a>Core Options</h3>
<p>
Here we include a more detailed description of the core options in the MOEDivCon
Score panel.
<p>
<object id="QBMOEDivConCore" title="QB MOEDivCon Core"
data="images/qb_core.png" type=image/png>
QB MOEDivCon Core
</object>

<h4><a name="core-options-outdb"></a>Output MOE Database</h4>
<p>
MOEDivCon uses a MOE database to store run parameters and to store
Score results; and if appropriate, analysis results.  
One advantage is that the database is self-contained:  all
of the data and parameters needed to execute MOEDivCon Score jobs are stored
in the database.  The database can be moved to a working directory prior to
execution time as long as the correct path is provided to the moebatch
command (see the <a href="#core-parallel-moebatch">Running Moebatch/MPI</a>
section for more information).  Another advantage is the suite of MOE
data analysis tools designed to work with MOE database.
<p>
<object id="QMScoreOutputDB" title="QMScore Output DataBase"
data="images/qmscoreoutputdb.png" type=image/png>
Output DataBase
</object>
<p>
MOEDivCon also creates an <inline>HDF5</inline> file in the same location and
with the same name as the MOE database (<inline>.h5</inline> extension).  This
file contains additional data (for instance, PWD) that the application uses for
post-execution analysis.

<h4><a name="core-options-workdir"></a>Working Directory</h4>
<p>
The working directory is used for storing QuantumBio DivCon intermediate
calculations and log files.  The path to the specified directory will
be stored in the MOEDivCon Score database and resolved at execution time.
If you are going to submit the MOEDivCon Score job on a cluster, you can 
specify the absolute path to a scratch directory mounted on the cluster's 
file system.  Each node involved in the parallel execution must have access
to the working directory.  If passed as a second argument to 
<inline>qmsmpi.svl</inline> in a cluster script, a subdirectory is created in
the working directory and MOEDivCon executes in it.
Also if MOEDivCon Score job is executed in parallel, 
a subdirectory under the working directory named <inline>logs</inline> in which
will be created a separate log file for each worker process.
<p>
An alternative to specifying a working directory is to use the <w>CWD</w>
option.
<p>
<object id="QMScoreCWD" title="QMScore CWD Option"
data="images/qmscorecwd.png" type=image/png>
QMScore CWD Option
</object>
<p>
In this case the current working directory will be set using the run time
environment variable <inline>$PWD</inline>.  This is effectively equivalent
to specifying <inline>'.'</inline> as the working directory.

<h4><a name="core-options-ham"></a>Hamiltonian</h4>
<p>
Several options are available in chosing the Hamiltonian algorithm to use
with each MOEDivCon Score calculation, each with its own default:  PM3 has 
proven to be the most reliable for QMScore and MNDO for NMRScore.

<h4><a name="core-options-scaling"></a>Scaling</h4>
<p>
Linear vs. Cubic referrs to the scaling factor of the calculation where
Linear scales accordingly to O(n) and Cubic scaling is the more conventional
variety (e.g. O(n&sup3;) or worse) where n corresponds to the number of
electrons in the system. The cross over point is at approximately 450 atoms.


<h2><a name="nmr"></a>MOEDivCon NMRScore</h2>
<p>

<h3><a name="nmr-intro"></a>Introduction</h3>
<p>

<p>
The patent pending NMRScore algorithm is a ligand binding mode scoring
function based upon DivCon's NMR chemical shift perturbation prediction
implementation. These CSP predictions are coupled with experimental CSP
values and becuase CSP is exquisitely sensitive on the orientation of the
ligand inside the binding pocket, NMRScore offers an accurate and
straightforward approach to score different poses.

<p>It has been 
<a href="http://www.quantumbioinc.com/publications/show/A-Critical-Assessment-of-the-Performance-of-Protein-Ligand-Scoring-Functions-Based-on-NMR-Chemical-Shift-Perturbations" target="_blank">demonstrated</a>
that the overall performance of NMRScore is superior to that of energy-based
scoring functions associated with docking programs in ranking native-like
structures and differentiating these poses from decoy poses. Therefore, by
combining conventional docking programs with NMRScore results in an approach
that can robustly determine the binding site structure for a protein-ligand
complex without having to resort to an expensive, time-comsuming xray
refinement.

<h3><a name="nmr-workflow"></a>Workflow</h3>
<p>
The NMRScore workflow is somewhat complicated by convergence issues due to the
sensitivity of the parameters of the Hamiltonians available for NMR 
calculations.
<p>
<object id="NMRScoreDefaults" title="NMRScore Panel"
data="images/nmrscoredefaults.png" type=image/png>
NMRScore Panel
</object>
<p>
<ol>
  <li>Prepare the receptor and poses for the NMRScore calculation.
  <li>Load the poses into a MOE database.
  <li>Open the receptor and one of the poses in MOE Window.
  <li>Use LigX to select the ligand in the receptor/pose complex
      (see <a href="#trouble-ligx"><w>instructions</w></a>).
  <li>Open the NMRScore panel:  
      <w>Compute</w>&rarr;<w>Simulations</w>&rarr;<w>QuantumBio</w>&rarr;<w>NMRScore</w>.
  <li>Enter the database name, working directory and select the
      <w>One-to-Many</w> mode.  Browse for the pose database.
  <li>Click <w>Create</w> to save the NMRScore database.
  <li>While the NMRScore database is still open in the MOE Database Viewer,
      click on <w>Perform</w> in the <w>Molecular Tests</w> section of the
      NMRScore panel.  If there are any errors or warnings, 
      use MOE's Atom Manager to find the problematic atoms (see the 
      <a href="#core-test">Molecule Tests</a> section).
      Go to <b>step 1</b>.
  <li>While the NMRScore database is still open in the MOE Database Viewer,
      click on <w>Check</w> in the <w>Charges</w> section of the
      NMRScore panel.  If there are any problematic charges, verify that
      the molecules are properly protonated.  If not, go to <b>step 1</b>.
      Otherwise,  enter the correct charge in the charge text boxes (select
      <w>All Poses</w> if appropriate).  Close the MOE Database Viewer and
      click <w>Update</w> to save the NMRScore database.  Recheck the charges by
      clicking <w>Check</w> again.  If problematic charges are reported in
      the <w>Status</w> field of the NMRScore database, you can edit the
      database.
      However, this should be the last step before running moebatch because if
      you save the NMRScore database by clicking <w>Save</w> your edits are
      overwritten.
      For further information see the <a href="#core-charge">Charges</a>
      section.
  <li>If you haven't already done so, 
      select the <w>Chemical Shift Atoms</w> and enter the experimental data
      in the edit panel (click <w>Enter/Edit Experimental Data</w>.
  <li>Close the MOE Database Viewer if it's still open and
      click <w>Save</w> to save the NMRScore database.  At this point, you
      may wish to make a backup copy of the database.
  <li>Run the NMRScore jobs using moebatch (see the 
      <a href="#core-parallel-moebatch">Running Moebatch/MPI</a> section).
  <li>After the jobs have completed, open the NMRScore database.  Verify that
      all the calculations converged ("Success" in the <w>Status</w> field).
      If not, close the MOE Database Viewer,
      open the NMRScore panel and click <w>Load</w> to load the NMRScore
      database (or backup copy of it).  You now have several options:
      <ul>
        <li>Cut the receptor. Goto  <b>step 7</b>.
        <li>Perform a geometry optimization on atoms in the pose.
            Goto  <b>step 11</b>.
        <li>Use precalculated density matrices to begin the calculations.
            Goto  <b>step 11</b>.
      </ul>
</ol>

<h3><a name="nmr-cut"></a>Receptor Cut</h3>
<p>
If convergence issues arise, one possible solution is to cut the receptor.
Once you have specified the radius,  all capped residues within the given
radius from the ligand are preserved.  You always have the option of
returning to original receptor by clicking on the <w>Original</w> button.

<h3><a name="nmr-csp"></a>Chemical Shift Atoms</h3>
<p>
In the study of protein/ligand interactions, one of the most common experimental
methods involves the determination of the chemical shift perturbation associated
with ligand binding. In this experiment, the shifts associated with the ligand
atoms (hydrogen and carbon) in both their bound and their unbound states are
determined. This shifts provide insight into the parts of the ligand that are
most affected by the binding site environment.
<p>
Recent 
<a href="http://www.quantumbioinc.com/publications/show//Support/Publications/A-Fast-QM-MM-Quantum-Mechanical-Molecular-Mechanical-Approach-to-Calculate-Nuclear-Magnetic-Resonance-Chemical-Shifts-for-Macromolecules" target="_blank">advances</a>
in linear scaling quantum mechanics have provided computational chemists with
the means to predict these shifts on a routine basis.

<h4><a name="nmr-csp-type"></a>Atom Type</h4>
<p>
There are currently three choices of atom types for the ligand atoms to undergo
chemical shift evaluation:
<ul>
  <li>H only:  only the hydrogen atoms in the ligand are scored.
  <li>C only:  only the carbon atoms in the ligand are scored.
  <li>all atoms: all atoms in the ligand are scored.
</ul>
<p>
<warn>Warning:  changing the CSP atom type deletes any experimental values 
entered previously.
</warn>

<h4><a name="nmr-csp-exp"></a>Experimental Data</h4>
<p>
Once you have selected the atom type, you need to supply the experimental
NMR data for those atoms.  When you click on the 
<w>Enter/Edit Experimental Data</w> button, a panel opens to allow you enter
or edit the data and the MOE Window zooms in on the ligand.  If you click on 
an atom label in the panel, the atom is labeled in the MOE Window.

<h3><a name="nmr-options"></a>Options</h3>
<p>
Here we include a more detailed description of the options in the MOEDivCon
NMRScore panel.

<h4><a name="nmr-options-opt"></a>Ligand Optimization</h4>
<p>
If a previous run of NMRScore did not converge, you may opt to have DivCon
perform a QM optimization of the ligand before attempting to calculate the
NMRScore.

<h4><a name="nmr-options-guess"></a>Guess</h4>
<p>
Perhaps the most useful methodology for speeding up DivCon is the use of the a
guess <a href="http://en.wikipedia.org/wiki/Density_matrix"
TARGET=_blank>density matrix<a/> derived from a previous QM characterization of
the same or similar protein structure. Whenever a calculation is performed, the
final density matrix of the simulation is saved to the HDF5 file. Without a
guess density matrix a unit matrix is used, so the addition of a guess can make
a signifcant impact on the calculation. In addition to the density matrix, the
current value of the shift value in eV is also saved.
<p>
<warn>Note:  not yet implemented.</warn>

<!--
<h3><a name="nmr-results"></a>Results</h3>
<p>
-->

<h2><a name="qms"></a>MOEDivCon QMScore</h2>
<p>

<h3><a name="qms-intro"></a>Introduction</h3>
<p>
Analogous to conventional scoring functions from other vendors, QuantumBio
provides a scoring function based upon linear-scaling, semi-empirical, quantum
mechanics. This patent pending algorithm characterizes the entire target/ligand
complex quantum mechanically, and determines the energy of binding.
<p>
The QMScore algorithm has been validated against several key structural motifs,
and the results of these validations have been detailed in several
key peer-reviewed
<a href="http://www.quantumbioinc.com/Support/Publications/index_files/category-qmscore.html"
   target="QBHomePage">publications</a>.


<h3><a name="qms-options"></a>Options</h3>
<p>
Here we include a more detailed description of the options in the MOEDivCon
QMScore panel.
<p>
<object id="QMScoreOptions" title="QMScore Options"
data="images/qmscoreoptions.png" type=image/png>
QMScore Options
</object>
<p>

<h4><a name="qms-options-opt"></a>Ligand Optimization</h4>
<p>
A QM-based ligand optimization can be performed prior to scoring.
This optimization may occur either while the ligand is docked within
the target or while the ligand is separate from the target.
Warning: this optional step will significantly increase the CPU cost
of the calculation.

<h4><a name="qms-options-shift"></a>Shift</h4>
<p>
User-defined initial dynamic level shift parameter in eV.

<h4><a name="qms-options-freq"></a>Frequency</h4>
<p>
Perform frequency calculation to determine ligand entropy in score
function.  Warning: may increase CPU cost of the calculation.

<h4><a name="qms-options-pwd"></a>PWD</h4>
<p>
Calculate the pairwise decomposition data needed for generating the Interaction
Heat Map and for the PLS analysis to generate the SAR Heat Map.
<p>
<warn>The QB-PWD analysis requires that the proteins have the same residue
numbers (see <a href="#pwd-align">Protein Alignment</a>).</warn>

<h3><a name="qms-results"></a>Results</h3>

<p>
The <a
href="http://www.quantumbioinc.com/publications?tag=qmscore"
TARGET="_blank">QMScore method</a>
has been validated on numerious structures and systems both
in academia and industry.<p>
<object id="ThermodynamicCycle" title="QMScore Thermodynamic Cycle"
data="images/thermodynamiccyle.png" type=image/png>
QMScore Thermodynamic Cycle
</object><ul>
    <li> dHf_g: Gas phase heat of formation. 
    <li> Solv_elec: The electrostatic portion of the solvation free energy. 
    <li> LJ_att:  Dispersion interactions from the attractive/dispersive part
         of the classical Lennard-Jones interaction potential.
    <li> S_solv:  Solvent component of the entropic contribution to binding.
    <li> S_vib:  Conformational component of the entropic contribution to binding.
    <li> E_eInt:  Electronic Interaction Energy
    <li> MRMScore:  Molecular Recognition Model score.
    <li> E_stericMRM:  Steric component of the Molecular Recognition Model score.
    <li> E_elecMRM:  Electrostatic component of the Molecular Recognition Model
         score.
    <li> E_solvMRM:  Pairwise solvation component of the Molecular Recognition
         Model score.
</ul>
<p>
<object id="AstexQMScoreResults" title="QMScore for Astex" width=956
data="images/astexmdb.png" type=image/png>
QMScore for Astex
</object>
<p>

<h2><a name="pwd"></a>MOEDivCon QB-PWD Analysis</h2>

<h3><a name="pwd-intro"></a>Introduction</h3>

<p>
The PairWise energy Decomposition (PWD) scheme for evaluating the electrostatic
interaction energy uses the neglect of nonbonded differential overlap (NNDO)
formalism applied to study protein-ligand interaction using the linear scaling
QM methods implemented within the DivCon package. This scheme permits the
calculation of the self-energy of the atom, core- electron interactions,
electron-electron repulsion, and exchange between atoms from the molecular
electron density.
<p>
Simulations using PWD yield raw atom-by-atom pairwise values which can be
difficult to interpret. Therefore, included with the tool is a receptor-based
QSAR method, comparative binding energy analysis (COMBINE), which is able to
process the data using statistical methods and provide a better picture of what
is happening within the active site. This method coupled with the semi-empirical
(SE) QM implementation within DivCon is referred to in the literature as
SE-COMBINE.
<p>
In addition to QM energy terms, molecular mechanics (MM) pairwise energy terms
are also provided derived from the Lennard-Jones potential. These terms, broken
into dispersive/attractive, repulsive, and electrostatics. When determining the
best scoring function for a system (see below), sometimes the best function is
one that combines the best of both QM and MM methods.
<p>
<warn>The current implementation of QB-PWD analysis can only handle 
cases where the ligand consists of a single "residue" (that is, all
the atoms have the same residue name and number).</warn>

<h3><a name="pwd-workflow"></a>Workflow</h3>
<p>
<object id="PWDDefaults" title="PWD Panel"
data="images/pwddefaults.png" type=image/png>
PWD Panel
</object>
<p>
The workflow of QB-PWD Analysis is captured in the following steps:
<ol>
  <li>Execute a batch run of QMScore with PWD selected on a set of receptor(s) 
      and poses.  The receptors must be aligned (see next section).
  <li>Open the QB-PWD panel:  
      <w>Compute</w>&rarr;<w>Simulations</w>&rarr;<w>QuantumBio</w>&rarr;<w>QB-PWD</w>.
  <li>Click on <w>Load</w> to open the QMScore MDB database containing the 
      results of the QMScore/QB-PWD batch run.
  <li>Select the options for the PWD analysis.  If you haven't done so already,
      load the experimental activities into the QMScore/QB-PWD database.
      Click <w>Apply</w>.
  <li>After the PLS analysis is complete, select a dataset.
  <li>Click <w>View</w>.  If the Interaction or SAR dataset is selected, 
      the surface of the protein pocket is shown in the MOE Window and a
      QB Heat Map is created to present the data.  Otherwise, a table
      of the selected dataset is shown.
  <li>Click <w>Save</w> to save the selected dataset in a 
      csv-formatted file.
</ol>

<h3><a name="pwd-align"></a>Protein Alignment</h3>
<p>
<object id="PWDProteinAlign" title="PWD Protein Align" width=956 
data="images/pwdproteinalign.png" type=image/png>
PWD Protein Align
</object>
<p>
In order to perform QB-PWD analysis on more than one protein, the residue
names and numbers must match;  that is, the proteins must be aligned.  You
are welcome to use your favorite tool, but we supply a script 
<inline>qbalign.svl</inline> in the MOEDivCon svl directory that performs
the task. The script reads in the protein of the receptor collections in 
the source directory, aligns them, renumbers the UIDs, and then 
stores the new receptor collections in the destination directory.  
Two assumptions are made:
<ul>
  <li>  The protein in the receptor collection is a single chain.
  <li>  The protein chain is the first chain in the receptor collection.
</ul>

There are two methods to execute the SVL script.  The first uses moebatch from a
shell command line in the parent directory of the source and destination
subdirectories.  The extension identifies the molecule file
type (supported extensions are pdb, moe, mol2, mdl_mol and macmod).
<pre>
&gt; moebatch -exec "run ['qbalign.svl', ['srcdir', 'destdir', 'ext']]" -exit
</pre>
An example:
<pre>
&gt; moebatch -exec "run ['qbalign.svl', ['recs', 'alignrecs', 'pdb']]" -exit
</pre>
The other method is to call the function from MOE's SVL command line (assuming
that moe was started in the parent directory of the source and destination
subdirectories.
<pre>
svl&gt; qb_Align ['srcdir', 'destdir', 'ext']
</pre>
Finally, create a MOE database containing the aligned proteins.

<h3><a name="pwd-analysis"></a>Analysis</h3>
<p>
There are several post analysis datasets that you can select:
<ul>
  <li>PLS Results Table:  a table that shows the following statistical results
      for each number of latent variables (1 to number of components) used in
      the  construction of the PLS models:  
    <ul>
      <li>X_Expl:  percent explanation of X variance
      <li>Act_Expl: percent explanation of activity
      <li>R2:  correlation coefficient (R<sup>2</sup>)
      <li>Q2: cross-validated correlation coefficient (Q<sup>2</sup>)
      <li>SDEC:  standard deviation of error of calculation
      <li>SDEP:  standard deviation of error of prediction
      <li>SDEP_Ext: standard deviation of error of prediction, external data set 
      <li>R2_Ext:  correlation coefficient, external data set
    </ul>
  <li>Interaction Heat Map:  a heat map that makes it easy to
      identify the residues that have the largest absolute effect on ligand
      binding.
  <li>SAR Heat Map:  a heat map that highlights residues that are most 
      critical to discriminating between more and less potent ligands.
</ul>

More will become available in the near future.

<h4><a name="pwd-analysis-view"></a>View</h4>
<p>
Click on the <w>View</w> button to view the selected dataset.  If the
Interaction or SAR dataset is selected, the surface of the protein pocket
is shown in the MOE Window and a QB Heat Map is created to present the data. 
Otherwise, a table of the selected dataset is shown.

<h4><a name="pwd-analysis-save"></a>Save</h4>
<p>
Click on the <w>Save</w> button to save the selected dataset to a csv-format
file, which is uses the filename 
<inline> &lt;Title&gt;-&lt;dataset&gt;.csv</inline> in the current working
directory.

<h3><a name="pwd-options"></a>Options</h3>
<p>
Here we include a more detailed description of the options in the MOEDivCon
QB-PWD panel.

<h4><a name="pwd-options-title"></a>Title</h4>
<p>
Enter a title for the QB-PWD analysis.  The title will be used to label the
Interaction and SAR Heat Maps and data tables, to form the filenames for 
datasets saved in csv-format, and to label the analytical
data in persistent storage.

<h4><a name="pwd-options-scorefun"></a>Scoring Function</h4>
<p>
The SE-COMBINE-based scoring functions utilize a mixture of the SE-QM PWD
methods of DivCon, and the MM PWD terms available from the Lennard-Jones
potential along with electrostatics terms available from Coulomb's law. These
physical terms are combined with the coefficients derived from the Partial
Least Squares (PLS) to yield the overall scoring function.
<p>
Note: when using the MOE interface, the MM terms provided are calculated using 
he currently selected MM force field within the MOE interface.
<ul>The terms of the QM and QM+MM scoring function(s) are broken down into the
    following QM elements:
    <li> E_AB:  the exchange term contributes negatively to binding and is 
         therefore a positive indicator of favorability.
    <li> E_ABp:  the electron-electron repulsion term contributes positively
         to binding and is therefore a negative indicator of favorability.
    <li> E_ABc:  the interactions between the core electrons.
</ul>

<ul>The terms of the MM and QM+MM scoring function(s) are broken down into the 
    following MM elements:
    <li> E_R6:  The r^6 term corresponds to the dispersive/attractive term of
         the Lennard-Jones potential.
    <li> E_R12: The r^12 term from the Lennard-Jones potential represents
         repulsion.
    <li> E_ELE:  Molecular mechanics electrostatics are represented using
         Coulomb's law along with MM charges.
</ul>

<h4><a name="pwd-options-residues"></a>Residues</h4>
<p>
In order to produce meaningful results during the PLS statistical analysis,
it is necessary to prune out residues that do not contribute any information.

<h5><a name="pwd-options-residues-pzero"></a>Pruning:  Percent Zero</h5>
<p>
Residues that are distant from the pocket have little if any interaction with
the poses and have mostly zero values for the energy terms in the scoring
function.  The inclusion of these residues in the analysis can distort the
PLS calculations.  Any residues with more than the given percentage of zero
energy values are pruned.  The default is set at 95 percent zero values.
<p>
Because of space limitations, the Heat Map truncates residues if there are
more than thirty of them, possibly losing important information.  If you
lower the threshhold for zero values (75%, say), more residues with lots of
zero values (and thus little interest) will be pruned during the analysis.

<h5><a name="pwd-options-residues-pSD"></a>Pruning:  Minimum SD</h5>
<p>
Residues that have the same interaction, whether weak or strong, with all of
the poses offer little information in discriminating one pose from another.
The PLS calculations are more reliable if these residues are not included in
the analysis.  Any residues with a smaller standard deviation of energy
values than the given SD are pruned. The default SD is set at 0.05.


<h5><a name="pwd-options-residues-ntop"></a>Number of Top Scoring Residues</h5>
<p>
Even with the pruning above, the majority of the remaining residues often
contribute little to the scoring function of the poses and offer little
information in the Interaction and SAR Heat Maps.  You can specify here the
number of top scoring residues to keep, which depends on the nature and size
of the protein pocket.
<p>
Note that the current implementation of QB Heat Map can show a maximum of
30 residues.

<h4><a name="pwd-options-pls"></a>PLS</h4>
<p>
Partial Least Squares models are built to describe the structure-activity of
the ligand poses.  Internal validation is carried out using Leave-One-Out (LOO)
cross-validation for the training set and the optimal dimensionality of each
model is assigned from its cross-validation predictive ability.  The optimal
PLS model is determined by the cross-validated correlation coefficient 
(Q<sup>2</sup>) values.


<h5><a name="pwd-options-pls-ncomps"></a>Number of Components</h5>
<p>
Enter the number of components to include in the model during the PLS analysis.


<h5><a name="pwd-options-pls-trainset"></a>Training Set Size</h5>
<p>
Enter the size of the training set.  The first N poses (Pose 1 ... Pose N) are
treated as the training set, and the remaining are treated as the prediction
set.  The receptor and ligand molecules can be ordered so that the training set
molecules appear first when the receptor and pose MDB databases are created, or
the entries in the databases can be rearranged after the database have been
created (see the MOE documentation).

<h2><a name="heatmap"></a>MOEDivCon Heat Map</h2>
<p>
The <w>QB Heat Map</w> is used to implement the interaction energy map and the
structure-activity relationship (SAR) map.  The <w>QB Heat Map</w> is 
interactive.  When first invoked, a Connolly molecular
surface of the pocket is generated.  Then, depending on the user selection, the
corresponding residues along the molecular surface are colored 
according to their interaction values in the heap map.
<p>
The interaction energy map is based
on the absolute <em>E</em><sub>AB</sub> terms from PWD, which  makes it
easy to identify the residues that have the largest absolute effect on ligand
binding.  Below is an example interaction heat map and associated surface for
ligand 4.
<p>
<object id="QBHeatMapInterPoly" title="QB Heat Map Interaction"
data="images/qbheatmapinterastexpoly.png" type=image/png>
QB Heat Map Interaction
</object>
<p>

<object id="SurfaceInterPoly" title="Surface Interaction" width=956
data="images/surfaceinterastexlig4poly.png" type=image/png>
Surface Interaction
</object>
<p>

The SAR map consists of the most important residue-ligand interaction
terms derived from the PLS analysis of all the interaction terms (attractive
and repulsive), which highlights residues that are most critical to discriminating
between more and less potent ligands (see the PWD section for more details).
Below is an example SAR heat map and associated surface for ligand 6.
<p>
<object id="QBHeatMapInterMono" title="QB Heat Map SAR"
data="images/qbheatmapsarastexmono.png" type=image/png>
QB Heat Map SAR
</object>
<p>
<object id="SurfaceSARMono" title="Surface SAR" width=956
data="images/surfacesarastexlig6mono.png" type=image/png>
Surface SAR
</object>
<p>
Note that ligand and residue names are abbreviated so that they fit into the
heat map.  If you hover the mouse over the abreviated names, a bubble appears
with the full name.  If you hover over a color-coded cell, the full corresponding 
ligand and residue names as well as the interaction or SAR value appears in 
the bubble.
<p>

<h3><a name="heatmap-selection"></a>Selection</h3>
<p>
There are three types of interactive selections you can make.  If you select a
ligand, the chosen ligand  is loaded into the MOE Window and the molecular
surface is recolored to show the values for all residues in the heat map for
that ligand.  If you select a residue, the ligand and surface do not
change, but the chosen residue is labeled and the view zooms in on it.  If
you select a color-coded cell, the corresponding ligand is loaded into the MOE 
Window and the molecular surface is recolored to show only the value for the
corresponding residue, which is also labeled.

<h3><a name="heatmap-saveprint"></a>Save and Print</h3>
<p>
You can save a publication quality image of the <w>QB Heat Map</w> by
clicking the <w>Save</w>  button.  MOE's <w>Graphic Export Panel</w> is used
to save the image.  Note that the default background is transparent.  If you
wish a white background, select the color white under the <w>Background</w>
option in the bottom right corner.  An example is shown below.
<p>
<object id="QBHeatMapSave" title="QB Heat Map Save"
data="images/qbheatmapsave.png" type=image/png>
QBPWD Heat Map Save
</object>
<p>
Likewise, MOE's <w>Print Preview Panel</w> is used to print
the <w>QB Heat Map</w> when you click the <w>Print</w> button.

<h3><a name="heatmap-animation"></a>Animation</h3>
<p>
When you click on the <w>Animate</w> button, the <w>QB Heat Map</w> is
temporarily hidden, and each ligand is sequentially selected beginning with
the topmost one.  That is, each ligand is loaded into the MOE Window and
the molecular surface is recolored.  When all ligands have been shown, the
<w>QB Heat Map</w> reappears.

<h3><a name="heatmap-setup"></a>Configuration</h3>
<p>
The <w>QB Heat Map</w> panel contains several options you can use to
configure the overall look of the heat map.  After changing the options,
click the <w>Apply</w> button to redraw the heat map and surface.
<p>
<object id="QBHeatmapConfig" title="QB Heatmap Config"
data="images/qbheatmapconfig.png" type=image/png>
QB Heatmap Config
</object>
<p>

<h4><a name="heatmap-setup-cutoff"></a>Cutoff</h4>
<p>
If <w>Cutoff</w> is selected, the standard deviation of a subset of the heat
map values is used to determine minimum and maximum cutoffs; otherwise, the
absolute minimum and maximum values are used.  In addition, you can specify
in the <w>Number of sima</w> field, how standard deviations are used to
determine the minimum and maximum values of the color scale.  Values smaller
than the minimum are assigned the same color as the minimum and
values greater than the maximum are assigned the same color as the maximum.
Not that for the Interaction Heat Map, it is usually better to not select a
cutoff.

<h4><a name="heatmap-setup-numcolors"></a>Number of Colors</h4>
<p>
The number of different colors used to generate the color scale for the heat map.

<h4><a name="heatmap-setup-mono"></a>Monochromatic</h4>
<p>
If <w>Monochromatic</w> is selected, a single color (red for positive,
blue for negative) is used to generate the color scale as was seen in the
SAR heat map above. Otherwise,
a rainbow spectrum of colors is used as was seen in the Interaction heat map
above.

<h4><a name="heatmap-setup-numticks"></a>Number of Ticks</h4>
<p>
The number of ticks to display on the color scale showing the values
represented by the colors.

<h2><a name="trouble"></a>Troubleshooting</h2>

We list here potential problems using the MOEDivCon integration and possible
solutions.  If your problem is not listed here or the proposed solution
doesn't help, please <a href="#contact">contact</a> support.
<ui>
  <li><p><prob>
    The QuantumBio submenu is not appearing in the MOE menu system.
    </prob>
    <p><sol>MOE requires that custom menu files
    (<inline>moe-menus</inline>) reside in the user's home directory.
    See the <a href="#install-menu">installation</a> section.
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
run: Environment variable not found.
    </pre>
    </prob><p><sol>
    MOE's <inline>run</inline> function is unable to find the MOEDivCon SVL
    scripts.  Make sure the <inline>MOE_SVL_RUNPATH</inline> is
    set up correctly (see the <a href="#install-path">installation</a>
    section).
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
    qb_QMScore: fwrite: Broken pipe.
    </pre>
    </prob><p><sol>
    MOE is unable to find the MOEDivCon executables.  Make sure the
    <inline>PATH</inline> is
    set up correctly (see the <a href="#install-path">installation</a>
    section).
  </sol>
  <li><p><prob>
    I get the message in the <w>Status</w> field of the QMScore database:
    <pre>
Error:   LICENSE ERROR: Problem with license file. Error Code: 1003.
    </pre>
    </prob><p><sol>
    MOEDivCon is distributed with an evaluation license which eventually
    expires.  See the <a href="#install-license">installation</a>
    section for more information on purchasing and installing a license.
  </sol>
  <li><p><prob>
    DivCon isn't generating a log file (<inline><em>divcon.log</em></inline>).
    </prob><p><sol>
    A logging configuration file <inline>divcon-log.config</inline> has to be
    in the working directory or in <inline>$QBHOME/etc</inline>.  The
    configuration file in the working directory takes precendence.
  </sol>
  <li><p><prob>
    MOEDivCon doesn't always include the hostname in the location for the 
    log file written in the <w>Status</w> field of the QMScore database.
    </prob><p><sol>
    Moebatch uses a ssh command to remotely execute MOEDivCon jobs, which
    doesn't always set the <inline>HOSTNAME</inline> environment variable.
    To ensure that the hostname always appears, add the following line to your 
    .bashrc file:
    <pre>
export HOSTNAME=`hostname`
    </pre>
    </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
"filename.mdb" is not a QMScore Database.
    </pre>
    </prob><p><sol>
    The MOE database that you are attempting to load isn't a QMScore database.
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
No molecular complex loaded in MOE Window.
    </pre>
    </prob><p><sol>
    When using QMScore in 'One-to-One' or 'One-to-Many' mode, you need to first
    open the molecule files (.pdb, .moe, etc.) of the complex (or receptor and
    ligand separately) into MOE.  MOEDivCon populates the QMScore database with
    the molecular data in the MOE Window.
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
No ligand was recognized.  Use LigX to define it.
    </pre>
    </prob><p><a name="trouble-ligx"></a><sol>
    For most protein/ligand complexes, MOE should be able to identify the ligand
    using its heuristics.  However in smaller systems, especially when the
    receptor has been cut, the ligand isn't always easily picked out.  You can
    assist the process with this quick guide to LigX.
    <ol>
      <li> Start LigX by pressing the <w>LigX</w> button in the right button
           bar.
      <li> Select an atom in the ligand.
      <li> Extend the selection to entire molecule by using the main or pop-up
           menu:
      <w>Selection</w> or <w>Select</w>&rarr;<w>Extend</w>&rarr;<w>Molecule</w>.
      <li> Tell LigX to mark the selected molecule as the ligand. In the right
           button bar, select <w>Prepare</w>&rarr;<w>Choose Ligand</w>.
      <li> Press the <w>Quit LigX</w> to exit from LigX.
      <li> Note that LigX alters the MOE View for pocket elucidation.  To 
           clean up the View, perform the following steps:
            <ol>
                <li> From the right button bar, 
                     select <w>Show</w>&rarr;<w>All</w>.
                <li> From the right button bar,
                     select <w>Color</w>&rarr;<w>Element</w>.
            </ol>
    </ol>
    MOEDivCon should now be able to identify the ligand.
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
Ligand has to be a single molecule.
    </pre>
    </prob><p><sol>
    The current version of QMScore expects one receptor and one ligand.  The
    receptor may comprise many chains, solvents, etc.  The ligand has to
    a single molecule.  If your problem involves multiple ligands within the
    same complex, we recommend you isolate each ligand for independent
    analysis.  Or <a href="#contact">contact</a> our scientific support group
    to resolve this issue in a future release.
  </sol>
    <p><sol>
    If you have edited the ligand (added or deleted atoms), the collection
    used to identify the ligand is invalid.  To fix the problem, you need to
    <ol>
        <li> From the SVL command line, execute
<pre>
svl>  oSetCollection ['QB_LIGAND', []]
</pre>
        <li> Use LigX to choose the ligand (see above).
    </ol>
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
Problematic molecule in receptor and/or ligand.  Continue anyway?
    </pre>
    </prob><p><sol>
    MOEDivCon has detected that there is a problem with the structure
    of either the receptor or ligand.  You should try to address the problem
    before running or saving the QMScore database.  MOEDivCon will skip the
    calculation of molecules with problematic structures.  You can learn
    what the problem is by clicking the <w>Perform</w> button in the
    <w>Molecule Tests</w> section with the "Error" tests selected.
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
Problematic charge for receptor and/or ligand.  Continue anyway?
    </pre>
    </prob><p><sol>
    MOEDivCon has detected that there is a problem with the charge
    of either the receptor or ligand.  The problematic molecule should be marked
    red in the <w>Charge</w> section.  You should try to address the problem
    before running or saving the QMScore database.  MOEDivCon will skip the
    calculation of molecules with problematic charges.
  </sol>
  <li><p><prob>
    In the <w>Status</w> field of the QMScore database, I get the message:
    <pre>
Error:  Problematic receptor and/or ligand.
    </pre>
or
    <pre>
Warning:  Problematic receptor and/or ligand.
    </pre>
    </prob><p><sol>
    You should get error or warning messages about a problematic molecules when 
    creating or saving a QMScore database (see the 
    <a href="#core-test">Molecule Tests</a> section for more information).
    You can right-click on the problematic molecule in the MOE Database Viewer,
    select <w>Send to MOE</w> from the popup menu, cleanup the molecule with
    MOE's suite of tools, then save it back in the database by right-clicking
    on the same molecule in the MOE  Database Viewer and
    selecting <w>Get from MOE</w> from the popup menu.  You may also wish to
    save the cleaned up molecule in its original format.
    <p>
    You can also use your favorite molecule editor to fix the problems using
    the original molecule files, then rebuild the QMScore database using the
    modified molecule files.
  </sol>
  <li><p><prob>
    In the <w>Status</w> field of the QMScore database, I get the message:
    <pre>
Error:  Problematic charge for receptor and/or ligand.
    </pre>
    </prob><p><sol>
    You should get a error messages about problematic charges when creating 
    or saving a QMScore database.  The preferred solution is to cleanup the
    molecule:  Right-click on the problematic molecule in the MOE Database
    Viewer,
    select <w>Send to MOE</w> from the popup menu, cleanup the molecule with
    MOE's suite of tools, then save it back in the database by right-clicking
    on the same molecule in the MOE  Database Viewer and
    selecting <w>Get from MOE</w> from the popup menu.  You may also wish to
    save the cleaned up molecule in its original format.
    <p>
    You can also use your favorite molecule editor to fix the problems using
    the original molecule files, then rebuild the QMScore database using the
    modified molecule files.
    <p>
    Or you can simply edit the charge in the MOE database (see the
    <a href="#core-charge-editing">charge</a> section).
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
The MOEDivCon database hasn't been created.
    </pre>
    </prob><p><sol>
    You are attempting to check the charges in the current QMScore database
    that hasn't been saved yet.  From the QMScore panel, ether save or load
    the QMScore database before continuing.
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
The MOEDivCon database isn't open.
    </pre>
    You are attempting to check the charges in the current QMScore database
    that isn't open.  From the QMScore panel, ether save or load
    the QMScore database before continuing.
    </prob><p><sol>
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
The pose (or receptor) database has more than one molecule field.
    </pre>
    </prob><p><sol>
    MOEDivCon cannot determine which is the correct molecule field for the
    poses (receptors) if there is more than one.  Copy the correct field into
    a new MOE database.
  </sol>
  <li><p><prob>
    I get the MOE message:
    <pre>
"filename1.mdb"  and "filename2.mdb"  are the same database.
    </pre>
    </prob><p><sol>
The QMScore database needs to a different database than the receptor and pose
databases.
  </sol>
  <li><p><prob>
    When setting up QMScore in <w>Many-to-Many</w> mode, I get the MOE message:
    <pre>
The receptor and pose databases have different number of entries.
    </pre>
    </prob><p><sol>
    In <w>Many-to-Many</w> mode, there should be a separate ligand/pose in the
    ligand/pose database for each receptor in the receptor database.  Verify
    that the databases specified in the QMScore panel are the correct ones.
  </sol>
  <li><p><prob>
    I submitted a parallel PBS (SGE, etc.) QMScore batch script and it returned
    immediately without executing any jobs.
    </prob><p><sol>
    There may be configuration problems.  Please <a href="#contact">contact</a>
    our support staff to resolve the issue.
  </sol>
  <li><p><prob>
    How do I know when all jobs of a MOEDivCon batch run have completed, or
    more importantly, are likely to converge.
    </prob><p><sol>
    Presently, MOEDivCon has no mechanism to show the user the progress of a
    calculation or whether it is converging.  This will be addressed in a future
    release of MOEDivCon.
  </sol>
    <li> <p><prob>
    I observe "atomic fusion" upon completion of a MOE minimization.
    </prob><p><sol>
    "Atomic fusion" or the movement of two atoms (usually
    protons) on top of each other has been observed when using the OPLS-AA force
    field within MOE. The problem is related to the fact that in OPLS-AA, polar
    hydrogens have zero van der Waals radii and that the default interior
    dielectric of the protein is 1. The problem can be addressed by turning on
    Born solvation at the start of the minimization. Unfortunately, there 
    doesn't seem to be a way to fix the problem once it has already occured. 
    Instead, you will need to go back to the original structure and re-optimize
    it.
    </sol>
</ui>

<h2><a name="contact"></a>Support Contact Information</h2>
<p>
<a href="http://www.quantumbioinc.com/Support/index.html"
   target="QBHomePageSupport"> <object height=51 align=left
   data="images/qb_logo.png" type=image/png title="QuantumBio Support Page">
QuantumBio Support Page
</object> </a>
<a href="http://www.quantumbioinc.com/Support/index.html"
   target="QBHomePageSupport"> <object id=ri height=51 align=right
   data="images/qb_logo.png" type=image/png title="QuantumBio Support Page">
QuantumBio Support Page
</object> </a>
<div id=ctr>
QuantumBio Inc.<br>
200 Innovation Blvd.<br>
State College, PA 16803<br>
Phone: 814-308-9406<br>
Fax: 814-235-6909<br>
email: support@quantumbioinc.com
</div>


<hr>
<p>&copy;2009-2010 by QuantumBio, Inc.

</body>

</html>
